signac/synced_collections/validators.py

```diff
@@ -69,6 +70,37 @@ def no_dot_in_key(data):
             no_dot_in_key(value)
 
 
+def require_string_key(data):
+    """Raise an exception if key in a mapping is not a string.
+
+    Almost all supported backends require string keys.
+
+    Parameters
+    ----------
+    data
+        Data to validate.
+
+    Raises
+    ------
+    KeyTypeError
+        If key type is not not a string.
```

Comment:
```suggestion
        If key type is not a string.
```

Reply:
...but what if the key is not not a string???